network_simulator/plot_tool.py

```diff
@@ -0,0 +1,300 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from statistics import *
+
+class PlotTool(object):
+    """
+    Produces graph of different data structures.
+    """
+    def __init__(self):
+        pass
+
+    def gen_count_interval_list(self, lst):
+        """
+        Computes a tuple list of counts based on a timestamp list.
+
+        :param list lst: list of timestamps to be converted.
+        :return: list
+        """
+        curr_count = 0
+        curr_time = GRAPH_WINDOW_SIZE
+        output = []
+        for idx in range(len(lst)):
+            timestamp = lst[idx]
+            if timestamp > curr_time:
+                output.append((curr_time, curr_count))
+                curr_count = 0
+                curr_time += GRAPH_WINDOW_SIZE
+            if timestamp <= curr_time:
+                curr_count += 1
+        # Append the last elements if they did not reach an end of an
+        # interval.
+        if curr_count != 0:
+            output.append((curr_time, curr_count))
+        return output
+
+    def gen_rate_tuple_list(self, tuple_list):
+        """
+        Computes a tuple list of rates based on a (timestamp, data) tuple
```

Comment:
Mention that the data always represents packet sizes (right?). Could just replace "data" with packet_size and refer to these rates as "flow rates".